File: src/main/java/com/cg/bookstoreapplication/dto/UserDTO.java
package com.cg.bookstoreapplication.dto;

import com.cg.bookstoreapplication.entities.Address;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class UserDTO {

	private int userId;
	private String firstName;
	private String lastName;
	private long phoneNumber;
	private String emailId;
	private Address address;
}

File: user-dto.ts
export class UserDTO {

    firstName: string;
    lastName: string;
    password: string;
    phoneNumber: string;
    gender: string;
    emailId: string;
    userRole:string;
    address:string;


    constructor(

        firstName: string,
        lastName: string,
        password: string,
        phoneNumber: string,
        gender: string,
        emailId: string,
        userRole:string,
        address:string

    ) {
        this.firstName = firstName;
        this.lastName = lastName;
        this.password = password;
        this.phoneNumber = phoneNumber;
        this.gender = gender;
        this.emailId = emailId;
        this.userRole=userRole;
        this.address=address;
    }

}
